PWGCF/DataModel/CorrelationsDerived.h

```diff
@@ -79,6 +79,48 @@ using CFTrack = CFTracks::iterator;
 using CFTrackLabel = CFTrackLabels::iterator;
 using CFTracksWithLabel = soa::Join<CFTracks, CFTrackLabels>;
 using CFTrackWithLabel = CFTracksWithLabel::iterator;
+
+//------transient CF-filter to CF-2prong-filter
+namespace cfcollref
+{
+DECLARE_SOA_INDEX_COLUMN(Collision, collision);
+} // namespace cfcollref
+DECLARE_SOA_TABLE(CFCollRefs, "AOD", "CFCOLLREF", o2::soa::Index<>, cfcollref::CollisionId); //! Transient cf collision index table
+
+using CFCollRef = CFCollRefs::iterator;
+
+namespace cftrackref
+{
+DECLARE_SOA_INDEX_COLUMN(Collision, collision);
+DECLARE_SOA_INDEX_COLUMN(Track, track);
+// DECLARE_SOA_INDEX_COLUMN(CFTrack, cfTrack); //! CF track index
+} // namespace cftrackref
+DECLARE_SOA_TABLE(CFTrackRefs, "AOD", "CFTRACKREF", o2::soa::Index<>, cftrackref::CollisionId, cftrackref::TrackId); //! Transient cf track index table
+// DECLARE_SOA_TABLE(CFTrackRefs, "AOD", "CFTRACKREF", track::CollisionId); //! Transient cf track index table
+
+using CFTrackRef = CFTrackRefs::iterator;
+//------
+
+namespace cf2prongtrack
+{
+DECLARE_SOA_INDEX_COLUMN(CFCollision, cfCollision);                                                                                                     //! Index to collision
+DECLARE_SOA_INDEX_COLUMN(CFMcParticle, cfMCParticle);                                                                                                   //! Index to MC particle
+DECLARE_SOA_INDEX_COLUMN_FULL(CFTrackProng0, cfTrackProng0, std::result_of<decltype (&aod::CFTrack::globalIndex)(aod::CFTrack)>::type, CFTracks, "_0"); //! Index to prong 1 track
+DECLARE_SOA_INDEX_COLUMN_FULL(CFTrackProng1, cfTrackProng1, std::result_of<decltype (&aod::CFTrack::globalIndex)(aod::CFTrack)>::type, CFTracks, "_1"); //! Index to prong 1 track
+// type
+DECLARE_SOA_COLUMN(Pt, pt, float);       //! pT (GeV/c)
+DECLARE_SOA_COLUMN(Eta, eta, float);     //! Pseudorapidity
+DECLARE_SOA_COLUMN(Phi, phi, float);     //! Phi angle
```

Comment:
can probably also be reused (from the definitions in this file)

Reply:
And if we want to use separate `Filter` for HF particles in the task?
Now I have
`Filter track2pFilter = (aod::cf2prongtrack::eta < cfgCutEta) && (aod::cf2prongtrack::pt > cfgCutPt)`
would it still be compatible with other filters if I use `::cftrack::` instead while reusing the columns?